src/ai-generator/infrastructure/api-client.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Yoast\WP\SEO\AI_Generator\infrastructure;
+
+use WPSEO_Utils;
+use Yoast\WP\SEO\AI_Generator\Infrastructure\Exceptions\WP_Request_Exception;
+
+/**
+ * Class API_Client
+ * Handles the API requests to the AI Generator API.
+ */
+class API_Client {
+
+	/**
+	 * The base URL for the API.
+	 *
+	 * @var string
+	 */
+	private $base_url;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $base_url The base URL for the API.
+	 */
+	public function __construct( string $base_url ) {
+		$this->base_url = $base_url;
+	}
+
+	/**
+	 * Performs a request to the API.
+	 *
+	 * @param string        $action_path The action path for the request.
+	 * @param array<string> $body        The body of the request.
+	 * @param array<string> $headers     The headers for the request.
+	 * @param bool          $is_post     Whether the request is a POST request.
+	 *
+	 * @return object The response from the API.
+	 *
+	 * @throws WP_Request_Exception When the wp_remote_post() returns an error.
+	 */
+	public function perform_request( string $action_path, $body, $headers, bool $is_post ): object {
+		$headers   = \array_merge( $headers, [ 'Content-Type' => 'application/json' ] );
+		$arguments = [
+			'timeout' => $this->get_request_timeout(),
+			'headers' => $headers,
+		];
+
+		if ( $is_post ) {
+			// phpcs:ignore Yoast.Yoast.JsonEncodeAlternative.Found -- Reason: We don't want the debug/pretty possibility.
+			$arguments['body'] = WPSEO_Utils::format_json_encode( $body );
+		}
+
+		/**
+		 * Filter: 'Yoast\WP\SEO\ai_api_url' - Replaces the default URL for the AI API with a custom one.
+		 **
+		 *
+		 * @since 21.0
+		 * @internal
+		 *
+		 * @param string $url The default URL for the AI API.
+		 */
+		$url      = \apply_filters( 'Yoast\WP\SEO\ai_api_url', $this->base_url );
+		$response = ( $is_post ) ? \wp_remote_post( $url . $action_path, $arguments ) : \wp_remote_get( $url . $action_path, $arguments );
+
+		if ( \is_wp_error( $response ) ) {
+			// phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped -- false positive.
+			throw new WP_Request_Exception( $response->get_error_message() );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Gets the timeout of the requests in seconds.
+	 *
+	 * @return int The timeout of the suggestion requests in seconds.
+	 */
+	public function get_request_timeout(): int {
```

Comment:
This should not have return type or a check that the filter returns an int and if its returns something else fallback on the default 60 seconds (which could also maybe be a constant).